src/Core/Content/ContactForm/SalesChannel/ContactFormRoute.php

```diff
@@ -135,10 +139,23 @@ public function load(RequestDataBag $data, SalesChannelContext $context): Contac
     private function validateContactForm(DataBag $data, SalesChannelContext $context): void
     {
         $definition = $this->contactFormValidationFactory->create($context);
+        $isPhoneOptionalInContactForm = $this->systemConfigService->get('core.basicInformation.isPhoneOptionalInContactForm', $context->getSalesChannel()->getId());
+
+        if ($isPhoneOptionalInContactForm === true) {
+            $definition->set('phone', new Optional());
+        }
+
+        $this->dispatchValidationEvent($definition, $context->getContext());
```

Comment:
:+1: for adding an event. I think extracting the dispatcher call into a method is not necessary here though.